Mathlib/Combinatorics/SimpleGraph/Bipartite.lean

```diff
@@ -0,0 +1,241 @@
+  rw [neighborFinset_def, ←disjoint_coe, Set.coe_toFinset]
+  exact isBipartite_neighborSet_disjoint h hv
+
+/-- If `G.IsBipartite s t` and `v ∈ s`, then the degree of `v` in `G` is at most the size of `t`. -/
+theorem isBipartite_degree_le (h : G.IsBipartite s t) (hv : v ∈ s) :
+    G.degree v ≤ #t := by
+  rw [←card_neighborFinset_eq_degree]
+  exact card_le_card (isBipartite_neighborFinset_subset h hv)
+
+/-- If `G.IsBipartite s t` and `w ∈ t`, then the neighbor finset of `w` is the set of vertices in
+`s` adjacent to `w` in `G`. -/
+theorem isBipartite_neighborFinset' (h : G.IsBipartite s t) (hw : w ∈ t) :
+    G.neighborFinset w = s.filter (G.Adj · w) := by
+  ext v
+  rw [mem_neighborFinset, adj_comm, mem_filter, iff_and_self]
+  exact h.mem_of_mem_adj' hw
+
+/-- If `G.IsBipartite s t` and `w ∈ t`, then the neighbor finset of `w` is the set of vertices
+"below" `w` according to the adjacency relation of `G`. -/
+theorem isBipartite_bipartiteBelow (h : G.IsBipartite s t) (hw : w ∈ t) :
+    G.neighborFinset w = bipartiteBelow G.Adj s w := by
+  rw [isBipartite_neighborFinset' h hw]
+  rfl
+
+/-- If `G.IsBipartite s t` and `w ∈ t`, then the neighbor finset of `w` is a subset of `s`. -/
+theorem isBipartite_neighborFinset_subset' (h : G.IsBipartite s t) (hw : w ∈ t) :
+    G.neighborFinset w ⊆ s := by
+  rw [isBipartite_neighborFinset' h hw]
+  exact filter_subset (G.Adj · w) s
+
+/-- If `G.IsBipartite s t` and `w ∈ t`, then the neighbor finset of `w` is disjoint to `t`. -/
+theorem isBipartite_neighborFinset_disjoint' (h : G.IsBipartite s t) (hw : w ∈ t) :
+    Disjoint (G.neighborFinset w) t := by
+  rw [neighborFinset_def, ←disjoint_coe, Set.coe_toFinset]
+  exact isBipartite_neighborSet_disjoint' h hw
+
+/-- If `G.IsBipartite s t` and `w ∈ t`, then the degree of `w` in `G` is at most the size of `s`. -/
+theorem isBipartite_degree_le' (h : G.IsBipartite s t) (hw : w ∈ t) :
+    G.degree w ≤ #s := by
+  rw [←card_neighborFinset_eq_degree]
+  exact card_le_card (isBipartite_neighborFinset_subset' h hw)
+
+/-- If `G.IsBipartite s t`, then the sum of the degrees of vertices in `s` is equal to the sum of
+the degrees of vertices in `t`.
+
+See `sum_card_bipartiteAbove_eq_sum_card_bipartiteBelow`. -/
+theorem isBipartite_sum_degrees_eq (h : G.IsBipartite s t) :
+    ∑ v ∈ s, G.degree v = ∑ w ∈ t, G.degree w := by
+  simp_rw [←sum_attach t, ←sum_attach s, ←card_neighborFinset_eq_degree]
+  conv_lhs =>
+    rhs; intro v
+    rw [isBipartite_bipartiteAbove h v.prop]
+  conv_rhs =>
+    rhs; intro w
+    rw [isBipartite_bipartiteBelow h w.prop]
+  simp_rw [sum_attach s fun w ↦ #(bipartiteAbove G.Adj t w),
+    sum_attach t fun v ↦ #(bipartiteBelow G.Adj s v)]
+  exact sum_card_bipartiteAbove_eq_sum_card_bipartiteBelow G.Adj
+
+variable [DecidableEq V]
+
+/-- The degree-sum formula only counting over the vertices that form edges.
+
+See `SimpleGraph.sum_degrees_eq_twice_card_edges`. -/
+theorem sum_degrees_support_eq_twice_card_edges :
```

Comment:
I think `sum_degrees_support_eq_twice_card_edges` should be in the same file as `sum_degrees_eq_twice_card_edges`